sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessage.java

```diff
@@ -57,6 +57,17 @@
 
     private Context context;
 
+    /**
+     * Creates a {@link ServiceBusMessage} with given byte array body.
+     *
+     * @param body The content of the Service bus message.
+     *
+     * @throws NullPointerException if {@code body} is null.
+     */
+    public ServiceBusMessage(byte[] body) {
+        this(BinaryData.fromBytes(Objects.requireNonNull(body, "'body' cannot be null.")));
```

Comment:
Functionally this works. But BinaryData.fromBytes() copies the byte array. Then when you get the byte array back from BinaryData, it's copied again. This will happen to every message. So this might not be the most optimal way.
Is it possible to use the `body` directly without the double copy?

Reply:
Althought .Net is doing the way you suggested., We talked about this with Jonathan as well and we want to keep it this way as long as we do drastically impact performance. 
Even earlier when only had byte[] , we were copying the array. Because we have a build time check for it. Also by having separate copy, we ensure in multi threaded, one thread is not changing data after creating SBM.
From performance point of view, I asked Vinay to test and see if there are some degradation of performance because of this and his data suggest there was no  degradation.

Reply:
No more concern if there is no impact to CPU performance and GC